### ISSUE TITLE
[Advise: The API call failed because the CUDA driver and runtime could not be initialized. ]

### ISSUE THREAD

**Issue by ryh95** [jina-ai/jina#2514]
**Describe your proposal/problem**

Dear Jina Team,

Recently, I tried `nlp-simple` example and replaced `TransformerTorchEncoder` with `TextPaddlehubEncoder`. However, I got the following error.

```
Traceback (most recent call last):
  File "/home/work/renyuanhang/my_anaconda3/envs/jina_rocket2/lib/python3.8/site-packages/jina/peapods/runtimes/zmq/zed.py", line 73, in _load_executor
    self._executor = BaseExecutor.load_config(
  File "/home/work/renyuanhang/my_anaconda3/envs/jina_rocket2/lib/python3.8/site-packages/jina/jaml/__init__.py", line 531, in load_config
    return JAML.load(tag_yml, substitute=False)
  File "/home/work/renyuanhang/my_anaconda3/envs/jina_rocket2/lib/python3.8/site-packages/jina/jaml/__init__.py", line 89, in load
    r = yaml.load(stream, Loader=JinaLoader)
  File "/home/work/renyuanhang/my_anaconda3/envs/jina_rocket2/lib/python3.8/site-packages/yaml/__init__.py", line 114, in load
    return loader.get_single_data()
  File "/home/work/renyuanhang/my_anaconda3/envs/jina_rocket2/lib/python3.8/site-packages/yaml/constructor.py", line 51, in get_single_data
    return self.construct_document(node)
  File "/home/work/renyuanhang/my_anaconda3/envs/jina_rocket2/lib/python3.8/site-packages/yaml/constructor.py", line 55, in construct_document
    data = self.construct_object(node)
  File "/home/work/renyuanhang/my_anaconda3/envs/jina_rocket2/lib/python3.8/site-packages/yaml/constructor.py", line 100, in construct_object
    data = constructor(self, node)
  File "/home/work/renyuanhang/my_anaconda3/envs/jina_rocket2/lib/python3.8/site-packages/jina/jaml/__init__.py", line 422, in _from_yaml
    return get_parser(cls, version=data.get('version', None)).parse(cls, data)
  File "/home/work/renyuanhang/my_anaconda3/envs/jina_rocket2/lib/python3.8/site-packages/jina/jaml/parsers/executor/legacy.py", line 130, in parse
    obj = cls(
  File "/home/work/renyuanhang/my_anaconda3/envs/jina_rocket2/lib/python3.8/site-packages/jina/executors/__init__.py", line 82, in __call__
    getattr(obj, '_post_init_wrapper', lambda *x: None)(m, r)
  File "/home/work/renyuanhang/my_anaconda3/envs/jina_rocket2/lib/python3.8/site-packages/jina/executors/__init__.py", line 174, in _post_init_wrapper
    self.post_init()
  File "/home/work/renyuanhang/my_anaconda3/envs/jina_rocket2/lib/python3.8/site-packages/jina/hub/encoders/nlp/TextPaddlehubEncoder/__init__.py", line 38, in post_init
    self.model = hub.Module(name=self.model_name)
  File "/home/work/renyuanhang/my_anaconda3/envs/jina_rocket2/lib/python3.8/site-packages/paddlehub/module/module.py", line 171, in __new__
    module = cls.init_with_name(
  File "/home/work/renyuanhang/my_anaconda3/envs/jina_rocket2/lib/python3.8/site-packages/paddlehub/module/module.py", line 263, in init_with_name
    user_module_cls = manager.install(name=name, version=version, source=source, update=update, branch=branch)
  File "/home/work/renyuanhang/my_anaconda3/envs/jina_rocket2/lib/python3.8/site-packages/paddlehub/module/manager.py", line 188, in install
    return self._install_from_name(name, version)
  File "/home/work/renyuanhang/my_anaconda3/envs/jina_rocket2/lib/python3.8/site-packages/paddlehub/module/manager.py", line 263, in _install_from_name
    return self._install_from_url(item['url'])
  File "/home/work/renyuanhang/my_anaconda3/envs/jina_rocket2/lib/python3.8/site-packages/paddlehub/module/manager.py", line 256, in _install_from_url
    return self._install_from_archive(file)
  File "/home/work/renyuanhang/my_anaconda3/envs/jina_rocket2/lib/python3.8/site-packages/paddlehub/module/manager.py", line 361, in _install_from_archive
    return self._install_from_directory(directory)
  File "/home/work/renyuanhang/my_anaconda3/envs/jina_rocket2/lib/python3.8/site-packages/paddlehub/module/manager.py", line 345, in _install_from_directory
    hub_module_cls = HubModule.load(self._get_normalized_path(module_info.name))
  File "/home/work/renyuanhang/my_anaconda3/envs/jina_rocket2/lib/python3.8/site-packages/paddlehub/module/module.py", line 219, in load
    paddle.set_device(place)
  File "/home/work/renyuanhang/my_anaconda3/envs/jina_rocket2/lib/python3.8/site-packages/paddle/device.py", line 166, in set_device
    framework._set_expected_place(place)
  File "/home/work/renyuanhang/my_anaconda3/envs/jina_rocket2/lib/python3.8/site-packages/paddle/fluid/framework.py", line 317, in _set_expected_place
    _set_dygraph_tracer_expected_place(place)
  File "/home/work/renyuanhang/my_anaconda3/envs/jina_rocket2/lib/python3.8/site-packages/paddle/fluid/framework.py", line 311, in _set_dygraph_tracer_expected_place
    _dygraph_tracer_._expected_place = place
OSError: (External)  Cuda error(3), initialization error.
  [Advise: The API call failed because the CUDA driver and runtime could not be initialized. ] (at /paddle/paddle/fluid/platform/gpu_info.cc:229)
```

It seems to be related with https://github.com/PaddlePaddle/Paddle/issues/25185. 

However, I have no idea how to fix this. Could you help me to figure it out? Thank you very much!

- - -

**Environment**

paddlehub==2.0.0
paddlenlp==2.0.1
paddlepaddle-gpu==2.0.2.post90
jina==1.3.0

**Screenshots**

![image](https://user-images.githubusercontent.com/8979086/120137462-f327db80-c206-11eb-8277-29deb3c6ef6d.png)

**Comment by LegendaryDan** [comment]
is there any update on this?

thanks.

**Comment by alexcg1** [comment]
@bwanglzu ?

**Comment by alexcg1** [comment]
Hey @ryh95 we've been discussing removing the `nlp-simple` example since it's pretty outdated (actually I may just do that now).

I'd suggest using `wikipedia-sentences` example instead and adapting from there. [It's being adapted for 2.0 as we speak](https://github.com/jina-ai/examples/pull/619). If you clone that and look at `transformer.py` you should be able to substitute in there.

Note: this may not actually solve the bug you're facing. But it will give you a more future-proofed experience with code that is  maintained.

Hope that helps!

**Comment by ryh95** [comment]
Hi, @alexcg1. I tried your suggestions, namely, I started from the `wikipedia-sentences` example and replaced the `TransformerTorchEncoder` with the `TextPaddlehubEncoder`. However, I still encountered the aforementioned `Cuda error(3)` with Jina 1.3

Maybe this problem is related to multiprocessing as [this issue](https://github.com/pytorch/pytorch/issues/2517) points out?

**Comment by jacobowitz** [comment]
Hi @ryh95 
I was just trying to reproduce it and for me it seems to work on my machine. Which cuda and PyTorch versions are you using? Can you please copy the output from nvidia-smi here?

I am on Ubuntu 20.04, torch==1.7.1

nvidia-smi yields:

 nvidia-smi 
```
Tue Jun  8 10:36:50 2021       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 465.19.01    Driver Version: 465.19.01    CUDA Version: 11.3     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  NVIDIA GeForce ...  Off  | 00000000:22:00.0 Off |                  N/A |
|  0%   53C    P0    31W / 160W |      0MiB /  5934MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
```

**Comment by jacobowitz** [comment]
I think your problem might be that you are using `paddlepaddle-gpu` . That is the version compiled with _cuda_ support but it appears you dont want to use a GPU here?
So I would recommend to just use paddlepaddle (not paddlepaddle-gpu)

**Comment by ryh95** [comment]
No, I want to use GPU.

The problem is it seems Jina 1.3 is not compatible with paddlepaddle-gpu.

**Comment by jacobowitz** [comment]
Im not sure that Jina 1.3 and paddlepaddle-gpu is necessarily a problem as it works for me. See my `pip freeze` package list below. Can you please paste your `nvidia-smi` output here?

```
aiofiles==0.7.0
appdirs==1.4.4
arrow==1.1.0
asgiref==3.3.4
astor==0.8.1
Babel==2.9.1
bce-python-sdk==0.8.60
binaryornot==0.4.4
certifi==2021.5.30
cfgv==3.3.0
chardet==4.0.0
click==7.1.2
colorama==0.4.4
colorlog==5.0.1
cookiecutter==1.7.3
cycler==0.10.0
decorator==5.0.9
dill==0.3.3
distlib==0.3.2
docker==5.0.0
easydict==1.9
fastapi==0.65.1
filelock==3.0.12
flake8==3.9.2
Flask==2.0.1
Flask-Babel==2.0.0
fluent-logger==0.10.0
future==0.18.2
gast==0.4.0
gevent==21.1.2
gitdb==4.0.7
GitPython==3.1.17
greenlet==1.1.0
grpcio==1.38.0
gunicorn==20.1.0
h11==0.12.0
h5py==3.2.1
identify==2.2.10
idna==2.10
itsdangerous==2.0.1
jieba==0.42.1
jina==1.3.0
Jinja2==3.0.1
jinja2-time==0.2.0
joblib==1.0.1
kiwisolver==1.3.1
lz4==3.1.1
MarkupSafe==2.0.1
matplotlib==3.4.2
mccabe==0.6.1
msgpack==1.0.2
multiprocess==0.70.11.1
nodeenv==1.6.0
numpy==1.20.3
opencv-python==4.5.2.54
packaging==20.9
paddlehub==2.0.0
paddlenlp==2.0.1
paddlepaddle-gpu==2.0.2
pandas==1.2.4
Pillow==8.1.2
pkg-resources==0.0.0
poyo==0.5.0
pre-commit==2.13.0
protobuf==3.17.2
pycodestyle==2.7.0
pycryptodome==3.10.1
pydantic==1.8.2
pyflakes==2.3.1
pyparsing==2.4.7
python-dateutil==2.8.1
python-magic==0.4.24
python-multipart==0.0.5
python-slugify==5.0.2
pytz==2021.1
PyYAML==5.4.1
pyzmq==22.1.0
rarfile==4.0
regex==2021.4.4
requests==2.25.1
scikit-learn==0.24.2
scipy==1.6.3
sentencepiece==0.1.95
seqeval==1.2.2
shellcheck-py==0.7.2.1
six==1.16.0
smmap==4.0.0
starlette==0.14.2
text-unidecode==1.3
threadpoolctl==2.1.0
toml==0.10.2
tornado==6.1
tqdm==4.61.0
typing-extensions==3.10.0.0
urllib3==1.26.5
uvicorn==0.14.0
uvloop==0.15.2
virtualenv==20.4.7
visualdl==2.2.0
websocket-client==1.0.1
websockets==9.1
Werkzeug==2.0.1
wsproto==1.0.0
zope.event==4.5.0
zope.interface==5.4.0
```

**Comment by ryh95** [comment]
Sure. The information that I think would be useful is as follows.

<details>
<summary> nvidia-smi </summary>

    Wed Jun  9 11:46:40 2021       
    +-----------------------------------------------------------------------------+
    | NVIDIA-SMI 384.81                 Driver Version: 384.81                    |
    |-------------------------------+----------------------+----------------------+
    | GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
    | Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
    |===============================+======================+======================|
    |   0  Tesla K40m          On   | 00000000:00:07.0 Off |                    0 |
    | N/A   31C    P0    62W / 235W |   8804MiB / 11439MiB |      0%      Default |
    +-------------------------------+----------------------+----------------------+
    |   1  Tesla K40m          On   | 00000000:00:08.0 Off |                    0 |
    | N/A   34C    P0    62W / 235W |   8075MiB / 11439MiB |      0%      Default |
    +-------------------------------+----------------------+----------------------+
    |   2  Tesla K40m          On   | 00000000:00:09.0 Off |                    0 |
    | N/A   33C    P0    61W / 235W |   7795MiB / 11439MiB |      0%      Default |
    +-------------------------------+----------------------+----------------------+
    |   3  Tesla K40m          On   | 00000000:00:0A.0 Off |                    0 |
    | N/A   46C    P0    63W / 235W |   9551MiB / 11439MiB |      0%      Default |
    +-------------------------------+----------------------+----------------------+
                                                                                   
    +-----------------------------------------------------------------------------+
    | Processes:                                                       GPU Memory |
    |  GPU       PID   Type   Process name                             Usage      |
    |=============================================================================|
    |    0     27905      C   python                                      8062MiB |
    |    0     31803      C   tools/python2.7_paddle_1.8.1/bin/python      729MiB |
    |    1     17954      C   python                                      8062MiB |
    |    2     13763      C   python                                       713MiB |
    |    2     16600      C   python                                      3612MiB |
    |    2     25810      C   python                                      3454MiB |
    |    3     12872      C   python                                      8544MiB |
    |    3     29961      C   ...gyu04/python2.7_paddle_1.8.5/bin/python   993MiB |
    +-----------------------------------------------------------------------------+

</details>

<details>
<summary>conda list</summary>

    # packages in environment at /home/work/renyuanhang/my_anaconda3/envs/jina_test:
    #
    # Name                    Version                   Build  Channel
    _libgcc_mutex             0.1                        main    defaults
    ca-certificates           2021.5.25            h06a4308_1    defaults
    certifi                   2021.5.30        py38h06a4308_0    defaults
    click                     8.0.1              pyhd3eb1b0_0    defaults
    cudatoolkit               9.0                  h13b8566_0    defaults
    cudnn                     7.6.5                 cuda9.0_0    defaults
    importlib-metadata        3.10.0           py38h06a4308_0    defaults
    ld_impl_linux-64          2.33.1               h53a641e_7    defaults
    libffi                    3.3                  he6710b0_2    defaults
    libgcc-ng                 9.1.0                hdf63c60_0    defaults
    libstdcxx-ng              9.1.0                hdf63c60_0    defaults
    ncurses                   6.2                  he6710b0_1    defaults
    openssl                   1.1.1k               h27cfd23_0    defaults
    pip                       21.1.1           py38h06a4308_0    defaults
    python                    3.8.10               hdb3f193_7    defaults
    readline                  8.1                  h27cfd23_0    defaults
    setuptools                52.0.0           py38h06a4308_0    defaults
    sqlite                    3.35.4               hdfb4753_0    defaults
    tk                        8.6.10               hbc83047_0    defaults
    wheel                     0.36.2             pyhd3eb1b0_0    defaults
    xz                        5.2.5                h7b6447c_0    defaults
    zipp                      3.4.1              pyhd3eb1b0_0    defaults
    zlib                      1.2.11               h7b6447c_3    defaults

</details>

<details>
<summary>pip freeze</summary>

    alabaster @ file:///home/ktietz/src/ci/alabaster_1611921544520/work
    anaconda-client==1.7.2
    anaconda-navigator==2.0.3
    anaconda-project @ file:///tmp/build/80754af9/anaconda-project_1610472525955/work
    anyio @ file:///tmp/build/80754af9/anyio_1617783275907/work/dist
    appdirs==1.4.4
    argh==0.26.2
    argon2-cffi @ file:///tmp/build/80754af9/argon2-cffi_1613037097816/work
    asn1crypto @ file:///tmp/build/80754af9/asn1crypto_1596577642040/work
    astor==0.8.1
    astroid @ file:///tmp/build/80754af9/astroid_1613500854201/work
    astropy @ file:///tmp/build/80754af9/astropy_1617745353437/work
    async-generator @ file:///home/ktietz/src/ci/async_generator_1611927993394/work
    atomicwrites==1.4.0
    attrs @ file:///tmp/build/80754af9/attrs_1604765588209/work
    autopep8 @ file:///tmp/build/80754af9/autopep8_1615918855173/work
    Babel @ file:///tmp/build/80754af9/babel_1607110387436/work
    backcall @ file:///home/ktietz/src/ci/backcall_1611930011877/work
    backports.functools-lru-cache @ file:///tmp/build/80754af9/backports.functools_lru_cache_1618170165463/work
    backports.shutil-get-terminal-size @ file:///tmp/build/80754af9/backports.shutil_get_terminal_size_1608222128777/work
    backports.tempfile @ file:///home/linux1/recipes/ci/backports.tempfile_1610991236607/work
    backports.weakref==1.0.post1
    bce-python-sdk==0.8.60
    beautifulsoup4 @ file:///home/linux1/recipes/ci/beautifulsoup4_1610988766420/work
    bitarray @ file:///tmp/build/80754af9/bitarray_1620827551536/work
    bkcharts==0.2
    black==19.10b0
    bleach @ file:///tmp/build/80754af9/bleach_1612211392645/work
    bokeh @ file:///tmp/build/80754af9/bokeh_1620779595936/work
    boto==2.49.0
    Bottleneck==1.3.2
    brotlipy==0.7.0
    certifi==2020.12.5
    cffi @ file:///tmp/build/80754af9/cffi_1613246945912/work
    cfgv==3.3.0
    chardet @ file:///tmp/build/80754af9/chardet_1607706746162/work
    click @ file:///home/linux1/recipes/ci/click_1610990599742/work
    cloudpickle @ file:///tmp/build/80754af9/cloudpickle_1598884132938/work
    clyent==1.2.2
    colorama @ file:///tmp/build/80754af9/colorama_1607707115595/work
    colorlog==5.0.1
    conda==4.10.1
    conda-build==3.21.4
    conda-content-trust @ file:///tmp/build/80754af9/conda-content-trust_1617045594566/work
    conda-package-handling @ file:///tmp/build/80754af9/conda-package-handling_1618262148928/work
    conda-repo-cli @ file:///tmp/build/80754af9/conda-repo-cli_1620168426516/work
    conda-token @ file:///tmp/build/80754af9/conda-token_1620076980546/work
    conda-verify==3.4.2
    contextlib2==0.6.0.post1
    cryptography @ file:///tmp/build/80754af9/cryptography_1616769286105/work
    cycler==0.10.0
    Cython @ file:///tmp/build/80754af9/cython_1618435160151/work
    cytoolz==0.11.0
    dask @ file:///tmp/build/80754af9/dask-core_1617390489108/work
    decorator==4.4.2
    defusedxml @ file:///tmp/build/80754af9/defusedxml_1615228127516/work
    diff-match-patch @ file:///tmp/build/80754af9/diff-match-patch_1594828741838/work
    dill==0.3.3
    distlib==0.3.2
    distributed @ file:///tmp/build/80754af9/distributed_1620902833129/work
    docutils @ file:///tmp/build/80754af9/docutils_1620827984873/work
    easydict==1.9
    entrypoints==0.3
    et-xmlfile==1.0.1
    faiss-cpu==1.7.1
    fastcache==1.1.0
    filelock @ file:///home/linux1/recipes/ci/filelock_1610993975404/work
    flake8 @ file:///tmp/build/80754af9/flake8_1615834841867/work
    Flask @ file:///home/ktietz/src/ci/flask_1611932660458/work
    Flask-Babel==2.0.0
    fsspec @ file:///tmp/build/80754af9/fsspec_1617959894824/work
    future==0.18.2
    gast==0.4.0
    gevent @ file:///tmp/build/80754af9/gevent_1616770671827/work
    gitdb==4.0.7
    GitPython==3.1.17
    glob2 @ file:///home/linux1/recipes/ci/glob2_1610991677669/work
    gmpy2==2.0.8
    greenlet @ file:///tmp/build/80754af9/greenlet_1611957705398/work
    grpcio==1.38.0
    gunicorn==20.1.0
    h5py==2.10.0
    HeapDict==1.0.1
    html5lib @ file:///tmp/build/80754af9/html5lib_1593446221756/work
    identify==2.2.10
    idna @ file:///home/linux1/recipes/ci/idna_1610986105248/work
    imageio @ file:///tmp/build/80754af9/imageio_1617700267927/work
    imagesize @ file:///home/ktietz/src/ci/imagesize_1611921604382/work
    importlib-metadata @ file:///tmp/build/80754af9/importlib-metadata_1617874469820/work
    iniconfig @ file:///home/linux1/recipes/ci/iniconfig_1610983019677/work
    intervaltree @ file:///tmp/build/80754af9/intervaltree_1598376443606/work
    ipykernel @ file:///tmp/build/80754af9/ipykernel_1596207638929/work/dist/ipykernel-5.3.4-py3-none-any.whl
    ipython @ file:///tmp/build/80754af9/ipython_1617120885885/work
    ipython-genutils @ file:///tmp/build/80754af9/ipython_genutils_1606773439826/work
    ipywidgets @ file:///tmp/build/80754af9/ipywidgets_1610481889018/work
    isort @ file:///tmp/build/80754af9/isort_1616355431277/work
    itsdangerous @ file:///home/ktietz/src/ci/itsdangerous_1611932585308/work
    jdcal==1.4.1
    jedi @ file:///tmp/build/80754af9/jedi_1606932564285/work
    jeepney @ file:///tmp/build/80754af9/jeepney_1606148855031/work
    jieba==0.42.1
    jina==1.3.0
    Jinja2 @ file:///tmp/build/80754af9/jinja2_1612213139570/work
    joblib @ file:///tmp/build/80754af9/joblib_1613502643832/work
    json5==0.9.5
    jsonschema @ file:///tmp/build/80754af9/jsonschema_1602607155483/work
    jupyter==1.0.0
    jupyter-client @ file:///tmp/build/80754af9/jupyter_client_1616770841739/work
    jupyter-console @ file:///tmp/build/80754af9/jupyter_console_1616615302928/work
    jupyter-core @ file:///tmp/build/80754af9/jupyter_core_1612213311222/work
    jupyter-packaging @ file:///tmp/build/80754af9/jupyter-packaging_1613502826984/work
    jupyter-server @ file:///tmp/build/80754af9/jupyter_server_1616083640759/work
    jupyterlab @ file:///tmp/build/80754af9/jupyterlab_1619133235951/work
    jupyterlab-pygments @ file:///tmp/build/80754af9/jupyterlab_pygments_1601490720602/work
    jupyterlab-server @ file:///tmp/build/80754af9/jupyterlab_server_1617134334258/work
    jupyterlab-widgets @ file:///tmp/build/80754af9/jupyterlab_widgets_1609884341231/work
    keyring @ file:///tmp/build/80754af9/keyring_1614616740399/work
    kiwisolver @ file:///tmp/build/80754af9/kiwisolver_1612282420641/work
    lazy-object-proxy @ file:///tmp/build/80754af9/lazy-object-proxy_1616526917483/work
    libarchive-c @ file:///tmp/build/80754af9/python-libarchive-c_1617780486945/work
    llvmlite==0.36.0
    locket==0.2.1
    lxml @ file:///tmp/build/80754af9/lxml_1616443220220/work
    lz4==3.1.1
    MarkupSafe==1.1.1
    matplotlib @ file:///tmp/build/80754af9/matplotlib-suite_1613407855456/work
    mccabe==0.6.1
    mistune==0.8.4
    mkl-fft==1.3.0
    mkl-random @ file:///tmp/build/80754af9/mkl_random_1618853849286/work
    mkl-service==2.3.0
    mock @ file:///tmp/build/80754af9/mock_1607622725907/work
    more-itertools @ file:///tmp/build/80754af9/more-itertools_1613676688952/work
    mpmath==1.2.1
    msgpack @ file:///tmp/build/80754af9/msgpack-python_1612287151062/work
    multipledispatch==0.6.0
    multiprocess==0.70.11.1
    mypy-extensions==0.4.3
    navigator-updater==0.2.1
    nbclassic @ file:///tmp/build/80754af9/nbclassic_1616085367084/work
    nbclient @ file:///tmp/build/80754af9/nbclient_1614364831625/work
    nbconvert @ file:///tmp/build/80754af9/nbconvert_1601914830498/work
    nbformat @ file:///tmp/build/80754af9/nbformat_1617383369282/work
    nest-asyncio @ file:///tmp/build/80754af9/nest-asyncio_1613680548246/work
    networkx @ file:///tmp/build/80754af9/networkx_1598376031484/work
    nltk @ file:///tmp/build/80754af9/nltk_1618327084230/work
    nodeenv==1.6.0
    nose @ file:///tmp/build/80754af9/nose_1606773131901/work
    notebook @ file:///tmp/build/80754af9/notebook_1616443462982/work
    numba @ file:///tmp/build/80754af9/numba_1616774046117/work
    numexpr @ file:///tmp/build/80754af9/numexpr_1618856167419/work
    numpy @ file:///tmp/build/80754af9/numpy_and_numpy_base_1618497241363/work
    numpydoc @ file:///tmp/build/80754af9/numpydoc_1605117425582/work
    olefile==0.46
    opencv-python==4.5.2.54
    openpyxl @ file:///tmp/build/80754af9/openpyxl_1615411699337/work
    packaging @ file:///tmp/build/80754af9/packaging_1611952188834/work
    paddlehub==2.0.0
    paddlenlp==2.0.1
    paddlepaddle-gpu==2.0.2.post90
    pandas==1.2.4
    pandocfilters @ file:///tmp/build/80754af9/pandocfilters_1605120460739/work
    parso==0.7.0
    partd @ file:///tmp/build/80754af9/partd_1618000087440/work
    path @ file:///tmp/build/80754af9/path_1614022220526/work
    pathlib2 @ file:///tmp/build/80754af9/pathlib2_1607024983162/work
    pathspec==0.7.0
    patsy==0.5.1
    pep8==1.7.1
    pexpect @ file:///tmp/build/80754af9/pexpect_1605563209008/work
    pickleshare @ file:///tmp/build/80754af9/pickleshare_1606932040724/work
    Pillow @ file:///tmp/build/80754af9/pillow_1617383569452/work
    pkginfo==1.7.0
    pluggy @ file:///tmp/build/80754af9/pluggy_1615976321666/work
    ply==3.11
    pre-commit==2.13.0
    prometheus-client @ file:///tmp/build/80754af9/prometheus_client_1618088486455/work
    prompt-toolkit @ file:///tmp/build/80754af9/prompt-toolkit_1616415428029/work
    protobuf==3.17.2
    psutil @ file:///tmp/build/80754af9/psutil_1612298023621/work
    ptyprocess @ file:///tmp/build/80754af9/ptyprocess_1609355006118/work/dist/ptyprocess-0.7.0-py2.py3-none-any.whl
    py @ file:///tmp/build/80754af9/py_1607971587848/work
    pycodestyle==2.7.0
    pycosat==0.6.3
    pycparser @ file:///tmp/build/80754af9/pycparser_1594388511720/work
    pycryptodome==3.10.1
    pycurl==7.43.0.6
    pydocstyle @ file:///tmp/build/80754af9/pydocstyle_1616182067796/work
    pyerfa @ file:///tmp/build/80754af9/pyerfa_1619390903914/work
    pyflakes==2.3.1
    Pygments @ file:///tmp/build/80754af9/pygments_1615143339740/work
    pylint @ file:///tmp/build/80754af9/pylint_1617135829881/work
    pyls-black @ file:///tmp/build/80754af9/pyls-black_1607553132291/work
    pyls-spyder @ file:///tmp/build/80754af9/pyls-spyder_1613849700860/work
    pyodbc===4.0.0-unsupported
    pyOpenSSL @ file:///tmp/build/80754af9/pyopenssl_1608057966937/work
    pyparsing @ file:///home/linux1/recipes/ci/pyparsing_1610983426697/work
    pyrsistent @ file:///tmp/build/80754af9/pyrsistent_1600141720057/work
    PySocks @ file:///tmp/build/80754af9/pysocks_1605305779399/work
    pytest==6.2.3
    python-dateutil @ file:///home/ktietz/src/ci/python-dateutil_1611928101742/work
    python-jsonrpc-server @ file:///tmp/build/80754af9/python-jsonrpc-server_1600278539111/work
    python-language-server @ file:///tmp/build/80754af9/python-language-server_1607972495879/work
    pytz @ file:///tmp/build/80754af9/pytz_1612215392582/work
    PyWavelets @ file:///tmp/build/80754af9/pywavelets_1601658317819/work
    pyxdg @ file:///tmp/build/80754af9/pyxdg_1603822279816/work
    PyYAML==5.4.1
    pyzmq==20.0.0
    QDarkStyle==2.8.1
    QtAwesome @ file:///tmp/build/80754af9/qtawesome_1615991616277/work
    qtconsole @ file:///tmp/build/80754af9/qtconsole_1616775094278/work
    QtPy==1.9.0
    rarfile==4.0
    redis==3.5.3
    regex @ file:///tmp/build/80754af9/regex_1617569202463/work
    requests @ file:///tmp/build/80754af9/requests_1608241421344/work
    rope @ file:///tmp/build/80754af9/rope_1602264064449/work
    Rtree @ file:///tmp/build/80754af9/rtree_1618420845272/work
    ruamel-yaml-conda @ file:///tmp/build/80754af9/ruamel_yaml_1616016699510/work
    scikit-image==0.18.1
    scikit-learn @ file:///tmp/build/80754af9/scikit-learn_1614446682169/work
    scipy @ file:///tmp/build/80754af9/scipy_1618855647378/work
    seaborn @ file:///tmp/build/80754af9/seaborn_1608578541026/work
    SecretStorage @ file:///tmp/build/80754af9/secretstorage_1614022784285/work
    Send2Trash @ file:///tmp/build/80754af9/send2trash_1607525499227/work
    seqeval==1.2.2
    shellcheck-py==0.7.2.1
    simplegeneric==0.8.1
    singledispatch @ file:///tmp/build/80754af9/singledispatch_1614366001199/work
    sip==4.19.13
    six @ file:///tmp/build/80754af9/six_1605205327372/work
    smmap==4.0.0
    sniffio @ file:///tmp/build/80754af9/sniffio_1614030475067/work
    snowballstemmer @ file:///tmp/build/80754af9/snowballstemmer_1611258885636/work
    sortedcollections @ file:///tmp/build/80754af9/sortedcollections_1611172717284/work
    sortedcontainers @ file:///tmp/build/80754af9/sortedcontainers_1606865132123/work
    soupsieve @ file:///tmp/build/80754af9/soupsieve_1616183228191/work
    Sphinx @ file:///tmp/build/80754af9/sphinx_1620777493457/work
    sphinxcontrib-applehelp @ file:///home/ktietz/src/ci/sphinxcontrib-applehelp_1611920841464/work
    sphinxcontrib-devhelp @ file:///home/ktietz/src/ci/sphinxcontrib-devhelp_1611920923094/work
    sphinxcontrib-htmlhelp @ file:///home/ktietz/src/ci/sphinxcontrib-htmlhelp_1611920974801/work
    sphinxcontrib-jsmath @ file:///home/ktietz/src/ci/sphinxcontrib-jsmath_1611920942228/work
    sphinxcontrib-qthelp @ file:///home/ktietz/src/ci/sphinxcontrib-qthelp_1611921055322/work
    sphinxcontrib-serializinghtml @ file:///home/ktietz/src/ci/sphinxcontrib-serializinghtml_1611920755253/work
    sphinxcontrib-websupport @ file:///tmp/build/80754af9/sphinxcontrib-websupport_1597081412696/work
    spyder @ file:///tmp/build/80754af9/spyder_1616775618138/work
    spyder-kernels @ file:///tmp/build/80754af9/spyder-kernels_1614030590686/work
    SQLAlchemy @ file:///tmp/build/80754af9/sqlalchemy_1620712430742/work
    statsmodels @ file:///tmp/build/80754af9/statsmodels_1614023746358/work
    sympy @ file:///tmp/build/80754af9/sympy_1618252284338/work
    tables==3.6.1
    tblib @ file:///tmp/build/80754af9/tblib_1597928476713/work
    terminado==0.9.4
    testpath @ file:///home/ktietz/src/ci/testpath_1611930608132/work
    textdistance @ file:///tmp/build/80754af9/textdistance_1612461398012/work
    threadpoolctl @ file:///tmp/tmp9twdgx9k/threadpoolctl-2.1.0-py3-none-any.whl
    three-merge @ file:///tmp/build/80754af9/three-merge_1607553261110/work
    tifffile==2020.10.1
    toml @ file:///tmp/build/80754af9/toml_1616166611790/work
    toolz @ file:///home/linux1/recipes/ci/toolz_1610987900194/work
    tornado @ file:///tmp/build/80754af9/tornado_1606942300299/work
    tqdm @ file:///tmp/build/80754af9/tqdm_1615925068909/work
    traitlets @ file:///home/ktietz/src/ci/traitlets_1611929699868/work
    typed-ast @ file:///tmp/build/80754af9/typed-ast_1610484547928/work
    typing-extensions @ file:///home/ktietz/src/ci_mi/typing_extensions_1612808209620/work
    ujson @ file:///tmp/build/80754af9/ujson_1611259522456/work
    unicodecsv==0.14.1
    urllib3 @ file:///tmp/build/80754af9/urllib3_1615837158687/work
    uvloop==0.15.2
    virtualenv==20.4.7
    visualdl==2.2.0
    watchdog @ file:///tmp/build/80754af9/watchdog_1612471027849/work
    wcwidth @ file:///tmp/build/80754af9/wcwidth_1593447189090/work
    webencodings==0.5.1
    Werkzeug @ file:///home/ktietz/src/ci/werkzeug_1611932622770/work
    widgetsnbextension==3.5.1
    wrapt==1.12.1
    wurlitzer @ file:///tmp/build/80754af9/wurlitzer_1617224664226/work
    xlrd @ file:///tmp/build/80754af9/xlrd_1608072521494/work
    XlsxWriter @ file:///tmp/build/80754af9/xlsxwriter_1617224712951/work
    xlwt==1.3.0
    xmltodict==0.12.0
    yapf @ file:///tmp/build/80754af9/yapf_1615749224965/work
    zict==2.0.0
    zipp @ file:///tmp/build/80754af9/zipp_1615904174917/work
    zope.event==4.5.0
    zope.interface @ file:///tmp/build/80754af9/zope.interface_1616357211867/work

</details>

<details>
<summary>echo $CUDA_VISIBLE_DEVICES</summary>

2

</details>

<details>
<summary>echo $LD_LIBRARY_PATH</summary>

/home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/:

</details>

**Comment by jacobowitz** [comment]
@ryh95 
Can you try paddlepaddle in your setup without jina? To make sure its not some other issue? I think this code should check that


```
import paddlehub as hub

model = hub.Module(name="ernie")
print('it worked')

```

**Comment by ryh95** [comment]
>>> import paddlehub as hub
    /home/work/renyuanhang/my_anaconda3/lib/python3.8/site-packages/paddle/fluid/layers/utils.py:26: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
    Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
      def convert_to_list(value, n, name, dtype=np.int):
    >>> model = hub.Module(name="ernie")
    /home/work/renyuanhang/my_anaconda3/lib/python3.8/site-packages/pip/_vendor/packaging/version.py:127: DeprecationWarning: Creating a LegacyVersion has been deprecated and will be removed in the next major release
      warnings.warn(
    /home/work/renyuanhang/my_anaconda3/lib/python3.8/site-packages/pip/_vendor/packaging/version.py:127: DeprecationWarning: Creating a LegacyVersion has been deprecated and will be removed in the next major release
      warnings.warn(
    [2021-06-10 17:11:09,017] [    INFO] - Already cached /home/work/.paddlenlp/models/ernie-1.0/ernie_v1_chn_base.pdparams
    W0610 17:11:09.019038  8688 device_context.cc:362] Please NOTE: device: 0, GPU Compute Capability: 3.5, Driver API Version: 9.0, Runtime API Version: 9.0
    W0610 17:11:09.029709  8688 device_context.cc:372] device: 0, cuDNN Version: 7.6.
    [2021-06-10 17:11:13,626] [    INFO] - Weights from pretrained model not used in ErnieModel: ['cls.predictions.layer_norm.weight', 'cls.predictions.decoder_bias', 'cls.predictions.transform.bias', 'cls.predictions.transform.weight', 'cls.predictions.layer_norm.bias']
    >>> print('it worked')
    it worked
I noticed that I haven't installed paddlehub in the previous post. I have installed it now and run the test you mentioned. However, the paddlehub seems ok and I have updated the information in the [previous post](https://github.com/jina-ai/jina/issues/2514#issuecomment-857353280).

**Comment by JoanFM** [comment]
Hey @ryh95 

May u try these 3 code snippets and let us know where it fails or which they fail and which they work?

Simple python
```
import paddlehub as hub
model = hub.Module(name='ernie')
```

Use our executor
```
from jina.hub.encoders.nlp.TextPaddlehubEncoder import TextPaddlehubEncoder
encoder = TextPaddlehubEncoder(model_name='ernie')
```

Use our executor distributed with a `Flow`:

```
from jina.hub.encoders.nlp.TextPaddlehubEncoder import TextPaddlehubEncoder
from jina.flow import Flow

f = Flow().add(uses=TextPaddlehubEncoder)
with f:
    pass
```

Knowing this would help us identify better the problem.

Thank you very much

**Comment by ryh95** [comment]
The first 2 works well. The last one fails.

    >>> import paddlehub as hub
    /home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/paddle/fluid/layers/utils.py:26: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
    Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
      def convert_to_list(value, n, name, dtype=np.int):
    >>> model = hub.Module(name='ernie')
    /home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/pip/_vendor/packaging/version.py:127: DeprecationWarning: Creating a LegacyVersion has been deprecated and will be removed in the next major release
      warnings.warn(
    /home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/pip/_vendor/packaging/version.py:127: DeprecationWarning: Creating a LegacyVersion has been deprecated and will be removed in the next major release
      warnings.warn(
    [2021-06-12 00:31:31,217] [    INFO] - Already cached /home/work/.paddlenlp/models/ernie-1.0/ernie_v1_chn_base.pdparams
    W0612 00:31:31.219398  8927 device_context.cc:362] Please NOTE: device: 0, GPU Compute Capability: 3.5, Driver API Version: 9.0, Runtime API Version: 9.0
    W0612 00:31:31.228368  8927 device_context.cc:372] device: 0, cuDNN Version: 7.6.
    [2021-06-12 00:31:35,728] [    INFO] - Weights from pretrained model not used in ErnieModel: ['cls.predictions.layer_norm.weight', 'cls.predictions.decoder_bias', 'cls.predictions.transform.bias', 'cls.predictions.transform.weight', 'cls.predictions.layer_norm.bias']
---
    >>> from jina.hub.encoders.nlp.TextPaddlehubEncoder import TextPaddlehubEncoder
    /home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/jina/jaml/parsers/__init__.py:84: DeprecationWarning: can not find parser for version: None, fallback to legacy parser. this usually mean you are using a depreciated YAML format.
      warnings.warn(
    /home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/jina/importer.py:266: UserWarning: due to the missing dependencies or bad implementations, ['jina.hub.encoders.image.FaceNetEncoder', 'jina.hub.segmenters.nlp.PDFPlumberSegmenter', 'jina.hub.encoders.nlp.CLIPTextEncoder', 'jina.hub.segmenters.image.FaceNetSegmenter', 'jina.hub.trainers.LightGBMRankerTrainer'] can not be imported if you are using these executors/drivers, they wont work. You can use `jina check` to list all executors and drivers
      warnings.warn(
    >>> encoder = TextPaddlehubEncoder(model_name='ernie')
    [2021-06-12 00:32:48,563] [    INFO] - Already cached /home/work/.paddlenlp/models/ernie-1.0/ernie_v1_chn_base.pdparams
    [2021-06-12 00:32:49,839] [    INFO] - Weights from pretrained model not used in ErnieModel: ['cls.predictions.layer_norm.weight', 'cls.predictions.decoder_bias', 'cls.predictions.transform.bias', 'cls.predictions.transform.weight', 'cls.predictions.layer_norm.bias']
---
    >>> from jina.hub.encoders.nlp.TextPaddlehubEncoder import TextPaddlehubEncoder
    >>> from jina.flow import Flow
    >>> f = Flow().add(uses=TextPaddlehubEncoder)
    >>> with f:
    ...     pass
    ...
               pod0@22379[I]:starting jina.peapods.runtimes.zmq.zed.ZEDRuntime...
               pod0@22379[I]:input tcp://0.0.0.0:57383 (PULL_BIND) output tcp://0.0.0.0:43449 (PUSH_BIND) control over tcp://0.0.0.0:41128 (PAIR_BIND)
    [2021-06-12 00:35:38,113] [ WARNING] - An error was encountered while loading ernie_tiny. Detailed error information can be found in the /home/work/.paddlehub/log/20210612.log.
    [2021-06-12 00:35:38,115] [ WARNING] - An error was encountered while loading ernie_tiny. Detailed error information can be found in the /home/work/.paddlehub/log/20210612.log.
            gateway@22382[I]:starting jina.peapods.runtimes.asyncio.grpc.GRPCRuntime...
    /home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/grpc/aio/_server.py:53: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
      self._server = cygrpc.AioServer(
            gateway@22382[I]:input tcp://0.0.0.0:43449 (PULL_CONNECT) output tcp://0.0.0.0:57383 (PUSH_CONNECT) control over
     ipc:///tmp/tmp4bd824co (PAIR_BIND)
            gateway@22382[S]:GRPCRuntime is listening at: 0.0.0.0:56724
    Download https://bj.bcebos.com/paddlehub/paddlehub_dev/ernie_tiny_2.0.2.tar.gz
    [##################################################] 100.00%
    Decompress /home/work/.paddlehub/tmp/tmp9b7_l5wq/ernie_tiny_2.0.2.tar.gz
    [##################################################] 100.00%
    [2021-06-12 00:35:55,054] [    INFO] - Successfully uninstalled ernie_tiny
               pod0@22379[C]:can not load the executor from TextPaddlehubEncoder
               pod0@22379[E]:ExecutorFailToLoad() during <bound method ZEDRuntime.setup of <jina.peapods.runtimes.zmq.zed.ZEDRuntime object at 0x7fbe13138640>>
     add "--quiet-error" to suppress the exception details
    Traceback (most recent call last):
      File "/home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/jina/peapods/runtimes/zmq/zed.py", line 73, in _load_executor
        self._executor = BaseExecutor.load_config(
      File "/home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/jina/jaml/__init__.py", line 531, in load_config
        return JAML.load(tag_yml, substitute=False)
      File "/home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/jina/jaml/__init__.py", line 89, in load
        r = yaml.load(stream, Loader=JinaLoader)
      File "/home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/yaml/__init__.py", line 114, in load
        return loader.get_single_data()
      File "/home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/yaml/constructor.py", line 51, in get_single_data
        return self.construct_document(node)
      File "/home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/yaml/constructor.py", line 55, in construct_document
        data = self.construct_object(node)
      File "/home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/yaml/constructor.py", line 100, in construct_object
        data = constructor(self, node)
      File "/home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/jina/jaml/__init__.py", line 422, in _from_yaml
        return get_parser(cls, version=data.get('version', None)).parse(cls, data)
      File "/home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/jina/jaml/parsers/executor/legacy.py", line 130, in parse
        obj = cls(
      File "/home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/jina/executors/__init__.py", line 82, in __call__
        getattr(obj, '_post_init_wrapper', lambda *x: None)(m, r)
      File "/home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/jina/executors/__init__.py", line 174, in _post_init_wrapper
        self.post_init()
      File "/home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/jina/hub/encoders/nlp/TextPaddlehubEncoder/__init__.py", line 38, in post_init
        self.model = hub.Module(name=self.model_name)
      File "/home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/paddlehub/module/module.py", line 171, in __new__
        module = cls.init_with_name(
      File "/home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/paddlehub/module/module.py", line 263, in init_with_name
        user_module_cls = manager.install(name=name, version=version, source=source, update=update, branch=branch)
      File "/home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/paddlehub/module/manager.py", line 188, in install
        return self._install_from_name(name, version)
      File "/home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/paddlehub/module/manager.py", line 263, in _install_from_name
        return self._install_from_url(item['url'])
      File "/home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/paddlehub/module/manager.py", line 256, in _install_from_url
        return self._install_from_archive(file)
      File "/home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/paddlehub/module/manager.py", line 361, in _install_from_archive
        return self._install_from_directory(directory)
      File "/home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/paddlehub/module/manager.py", line 345, in _install_from_directory
        hub_module_cls = HubModule.load(self._get_normalized_path(module_info.name))
      File "/home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/paddlehub/module/module.py", line 219, in load
        paddle.set_device(place)
      File "/home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/paddle/device.py", line 166, in set_device
        framework._set_expected_place(place)
      File "/home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/paddle/fluid/framework.py", line 317, in _set_expected_place
        _set_dygraph_tracer_expected_place(place)
      File "/home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/paddle/fluid/framework.py", line 311, in _set_dygraph_tracer_expected_place
        _dygraph_tracer_._expected_place = place
    OSError: (External)  Cuda error(3), initialization error.
      [Advise: The API call failed because the CUDA driver and runtime could not be initialized. ] (at /paddle/paddle/fluid/platform/gpu_info.cc:229)
    
    
    The above exception was the direct cause of the following exception:
    
    Traceback (most recent call last):
      File "/home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/jina/peapods/peas/__init__.py", line 71, in run
        self.runtime.setup()
      File "/home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/jina/peapods/runtimes/zmq/zed.py", line 52, in setup
        self._load_executor()
      File "/home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/jina/peapods/runtimes/zmq/zed.py", line 93, in _load_executor
        raise ExecutorFailToLoad from ex
    jina.excepts.ExecutorFailToLoad
               pod0@8927[S]:terminated
               Flow@8927[E]:pod0:<jina.peapods.pods.Pod object at 0x7fbe1311bfd0> can not be started due to RuntimeFailToStart(), Flow is aborted
            gateway@22382[I]:#sent: 0 #recv: 0 sent_size: 0 Bytes recv_size: 0 Bytes
    /home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/asyncio/events.py:81: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
      self._context.run(self._callback, *self._args)
            gateway@8927[S]:terminated
               Flow@8927[S]:flow is closed and all resources are released, current build level is EMPTY
               Flow@8927[S]:flow is closed and all resources are released, current build level is EMPTY
    Traceback (most recent call last):
      File "<stdin>", line 1, in <module>
      File "/home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/jina/flow/base.py", line 514, in __enter__
        return self.start()
      File "/home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/jina/flow/base.py", line 559, in start
        v.wait_start_success()
      File "/home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/jina/peapods/pods/__init__.py", line 543, in wait_start_success
        p.wait_start_success()
      File "/home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/jina/peapods/peas/__init__.py", line 145, in wait_start_success
        raise RuntimeFailToStart
    jina.excepts.RuntimeFailToStart

**Comment by JoanFM** [comment]
may u share the output of the log file from paddle?
/home/work/.paddlehub/log/20210612.log ?

**Comment by ryh95** [comment]
the running result and the log are as follows:

    /home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/paddle/fluid/layers/utils.py:26: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
    Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
      def convert_to_list(value, n, name, dtype=np.int):
    /home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/pip/_vendor/packaging/version.py:127: DeprecationWarning: Creating a LegacyVersion has been deprecated and will be removed in the next major release
      warnings.warn(
    /home/work/renyuanhang/my_anaconda3/envs/jina_test/lib/python3.8/site-packages/pip/_vendor/packaging/version.py:127: DeprecationWarning: Creating a LegacyVersion has been deprecated and will be removed in the next major release
      warnings.warn(
    [2021-06-12 12:44:06,543] [    INFO] - Already cached /home/work/.paddlenlp/models/ernie-1.0/ernie_v1_chn_base.pdparams
    W0612 12:44:06.545397  7495 device_context.cc:362] Please NOTE: device: 0, GPU Compute Capability: 3.5, Driver API Version: 9.0, Runtime API Version: 9.0
    W0612 12:44:06.553900  7495 device_context.cc:372] device: 0, cuDNN Version: 7.6.
    [2021-06-12 12:44:11,101] [    INFO] - Weights from pretrained model not used in ErnieModel: ['cls.predictions.layer_norm.weight', 'cls.predictions.decoder_bias', 'cls.predictions.transform.bias', 'cls.predictions.transform.weight', 'cls.predictions.layer_norm.bias']

[20210612.log](https://github.com/jina-ai/jina/files/6641655/20210612.log)

**Comment by JoanFM** [comment]
so this last proposal worked then?

**Comment by ryh95** [comment]
Yes

**Comment by JoanFM** [comment]
hey @ryh95 ,

Looking at ur `nvidia-smi` output, it strikes me the process it seems to be running:

```tools/python2.7_paddle_1.8.1/bin/python```

I have 2 questions, are you sure to be running jina with `python` version above 3.7? is it possible that this process already initialized CUDA and that's why u cannot set another one somehow?

**Comment by ryh95** [comment]
I run jina with python 3.8 which is installed in a conda environment called jina_test.
The following is my python version.

    (jina_test) [work@szwg-gpu-199-240-34-00.epc wikipedia-sentences]$which python
    ~/renyuanhang/my_anaconda3/envs/jina_test/bin/python
    (jina_test) [work@szwg-gpu-199-240-34-00.epc wikipedia-sentences]$~/renyuanhang/my_anaconda3/envs/jina_test/bin/python
    Python 3.8.10 (default, May 19 2021, 18:05:58)
    [GCC 7.3.0] :: Anaconda, Inc. on linux
    Type "help", "copyright", "credits" or "license" for more information.

This can also be proved by the previous results and 20210612.log.
Since this is a server used by a lot of people, the process `tools/python2.7_paddle_1.8.1/bin/python` should be someone else's. If I have initialized the CUDA process. Then there should exist a process named `~/renyuanhang/my_anaconda3/envs/jina_test/bin/python`
On the other hand, I checked these processes. These are not my processes.

**Comment by bwanglzu** [comment]
hi @ryh95 does the above solution work for you? Seems it's an pytorch + multiprocessing issue. 

if not , can you try:

```python
import os
os.environ["CUDA_DEVICE_ORDER"]="PCI_BUS_ID"
os.environ["CUDA_VISIBLE_DEVICES"]=""
```

**Comment by ryh95** [comment]
Hi, @bwanglzu 

Thank you for your script, the error changed to another one which is irrelevant with cuda.
I have 2 questions.
1. Could you explain why this os config can work?
2. What if I want to specify a gpu card with this config? I tried to set CUDA_VISIBLE_DEVICES with above script but it fails.

**Comment by bwanglzu** [comment]
@ryh95 i just found this [link](https://github.com/PaddlePaddle/PaddleOCR/issues/3070#issuecomment-857698911), can you check if it works for you? I don't have a gpu on my laptop and can not reproduce.

**Comment by ryh95** [comment]
@bwanglzu 
Thanks for your advice!
But I don't use torch in this issue. Sorry for the misleading.

**Comment by bwanglzu** [comment]
hi @ryh95 I'm a bit confused by the current status, is it resolved, by Joan's latest comments on multiprocessing, or your program still have new errors popped up?

**Comment by ryh95** [comment]
No, the original cuda error (3) is not solved. There are no new errors.

**Comment by bwanglzu** [comment]
hi @ryh95 i looked into the issue, seems it's not only related to Paddle, but also Pytorch, look at this [thread](https://github.com/PaddlePaddle/Paddle/issues/20169#issuecomment-557990722). Basically, Jina use multithreading with `fork` as start method, while CUDA, and for Jina we can not switch to `spawn`.

As far as I know, serving your solution without enabling GPU is the easier way.

**Comment by ryh95** [comment]
Ok, Would you support other starting methods in the future?

**Comment by bwanglzu** [comment]
@ryh95 We're currently having a discussion, I'll get you updated if we make any decision. Btw, are you interested in upgrade your app with Jina 2.0? Jina 1.3 is a bit non-trival to use.

**Comment by ryh95** [comment]
Yes, I've been using 2.0 right now.

**Comment by JoanFM** [comment]
Hello @ryh95 ,

We believe that the problem is related to the way we start new process, and we think that if we'd allow `spawn` to be the process starter method, the problem would be fixed.

Would you be able to confirm it by trying to reproduce your issue running on this branch `fix-allow-spawn` (https://github.com/jina-ai/jina/pull/2688)

This would help us be more sure if the diagnosis is correct, and if it is, to push forward int his direction.

Thank you very much

**Comment by ryh95** [comment]
Hi, @JoanFM 
I switched the branch to the `fix-allow-spawn` and installed Jina with `setup.py`.
But it seems it's a 2.0 version and I can not run my wikipedia-sentences example now.
Could you give me a script that I can run?
I'll tell you the result then.

**Comment by JoanFM** [comment]
Hey @ryh95 ,

would you be able to run the following snippet?

```python
from jina import Flow, Executor, requests
import paddlehub as hub


class PaddleHubExecutor(Executor):

    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.model = hub.Module(name='ernie')
        print(f' \n\n\n SUCCESS in loading model \n\n\n')

    @requests
    def encode(self, docs, **kwargs):
        print(f' here I would encode')


f = Flow().add(uses=PaddleHubExecutor)
with f:
    pass


```

**Comment by JoanFM** [comment]
Wait @ryh95 , I need to fix a couple of things

**Comment by JoanFM** [comment]
Hey @ryh95 ,

Can u please try this one (on the branch):

```python
from jina import Flow, Executor, requests
import paddlehub as hub


class PaddleHubExecutor(Executor):

    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.model = hub.Module(name='ernie')
        print(f' \n\n\n SUCCESS in loading model \n\n\n')

    @requests
    def encode(self, docs, **kwargs):
        print(f' here I would encode')


if __name__ == '__main__':
    from multiprocessing import set_start_method as _set_start_method

    _set_start_method('spawn')

    f = Flow().add(uses=PaddleHubExecutor)
    f.use_rest_gateway()
    with f:
        pass
```

**Comment by ryh95** [comment]
Hi, @JoanFM 
The model is loaded successfully!
No errors are detected.
Thank you for your work!

**Comment by JoanFM** [comment]
Hey @ryh95 

So this gives us more confidence of the underlying problem. It is not a trivial problem to fix but we will try to work on it asap.

I will close this issue and link it to a new one where the `underlying` problem can be managed

**Comment by JoanFM** [comment]
Follow-up ticket opened at #2691

**Comment by bwanglzu** [comment]
hi @ryh95 can you print your cuda version?

```shell
nvcc --version
```

**Comment by ryh95** [comment]
nvcc: NVIDIA (R) Cuda compiler driver
    Copyright (c) 2005-2017 NVIDIA Corporation
    Built on Fri_Sep__1_21:08:03_CDT_2017
    Cuda compilation tools, release 9.0, V9.0.176

**Comment by bwanglzu** [comment]
okay, actually we're having testing scripts using pytorch with gpu enabled (with CUDA 11+), and it worked. We discuss how to resolve the issue, thanks so much!

**Comment by ryh95** [comment]
So, cuda 11+ supports the ` fork` method?

**Comment by bwanglzu** [comment]
yes, or at least did not break in our stress test.

**Comment by JoanFM** [comment]
Hey @ryh95,

Right now with `2.0.0.rc8` you should be able to work if u manually set `spawn` as the `start method` for processes as in the code snippet we shared.

**Comment by JoanFM** [comment]
Not true @ryh95 sorry for the inconvenience, not working yet

**Comment by hanxiao** [comment]
This should be solved in https://github.com/jina-ai/jina/pull/3062

@ryh95 could you please do the following

- `pip install --pre jina` (update Jina to latest master, i.e. prerelease)
- set env `JINA_MP_START_METHOD="spawn"` in the console session **before starting Jina & Python**?
   ```bash
   export JINA_MP_START_METHOD=spawn
   python your_jina_app.py
   ```
   or 
   ```bash
   JINA_MP_START_METHOD=spawn python your_jina_app.py
   ```
  either should work.

A reproducible example can be found in the PR body of https://github.com/jina-ai/jina/pull/3062
